scripts/migrations/migrate_2.3.0.js

```diff
@@ -0,0 +1,106 @@
+const shell = require("shelljs");
+const { readContracts } = require("../util/utils.js");
+const hre = require("hardhat");
+const ethers = hre.ethers;
+const network = hre.network.name;
+// const { getStateModifyingFunctionsHashes } = require("../../scripts/util/diamond-utils.js");
+const tag = "HEAD";
+const version = "2.3.0";
+
+const config = {
+  // status at 451dc3d. ToDo: update this to the latest commit
+  addOrUpgrade: [
+    "DisputeResolverHandlerFacet",
+    "FundsHandlerFacet",
+    "MetaTransactionsHandlerFacet",
+    "OfferHandlerFacet",
+    "OrchestrationHandlerFacet1",
+    "ProtocolInitializationHandlerFacet",
+    "SellerHandlerFacet",
+    "TwinHandlerFacet",
+  ],
+  remove: [],
+  skipSelectors: {},
+  facetsToInit: {},
+  initializationData: "0x",
+};
+
+async function migrate(env) {
+  console.log(`Migration ${tag} started`);
+  try {
+    console.log("Removing any local changes before upgrading");
+    shell.exec(`git reset @{u}`);
+    const statusOutput = shell.exec("git status -s -uno scripts");
+
+    if (statusOutput.stdout) {
+      throw new Error("Local changes found. Please stash them before upgrading");
+    }
+
+    if (env != "upgrade-test") {
+      console.log("Installing dependencies");
+      shell.exec(`npm install`);
+    }
```

Comment:
This logic is the same that we have on migrate_2.2.1 (and looks like a thing that we need to check on every migration) so can you put this in a shared function so we can avoid duplication?

Reply:
I know. That's why I put this in the PR description
> NB: This is not the final version of the migration script. It will be adapted as part of https://github.com/bosonprotocol/boson-protocol-contracts/pull/680 when other PRs are merged into the main. The parts that are common with migrate_2.2.1 will be refactored into util functions.